server/executor/integration_test.go

```diff
@@ -24,25 +24,87 @@ func TestExecutorIntegration(t *testing.T) {
 	require.NoError(t, err)
 	defer demoApp.Stop()
 
-	app, err := test.GetTestingApp(demoApp)
+	tracetestApp, err := test.GetTestingApp(demoApp)
 	require.NoError(t, err)
 
-	go app.Start()
+	go tracetestApp.Start()
 
 	time.Sleep(1 * time.Second)
 
+	testCases := []struct {
+		Name string
+		Fn   func(*testing.T, *app.App, *test.DemoApp)
+	}{
+		{
+			Name: "Test with new selector should be asserted by assertion engine",
+			Fn:   happyPathWithNewSelector,
```

Comment:
I thought about that, but that would increase the complexity of the setup because I would have to ensure only one stack is being called when `setup()` is called twice, otherwise, this means we will have to start everything twice and that would add a 30s overhead on the tests.

Reply:
In that case, I would use direct function call instead of a loop simulating a table test.